agent/api/ecsclient/client.go

```diff
@@ -325,12 +327,29 @@ func validateRegisteredAttributes(expectedAttributes, actualAttributes []*ecs.At
 }
 
 func (client *APIECSClient) getAdditionalAttributes() []*ecs.Attribute {
-	return []*ecs.Attribute{
+	attrs := []*ecs.Attribute{
 		{
-			Name:  aws.String("ecs.os-type"),
+			Name:  aws.String(osTypeAttrName),
 			Value: aws.String(config.OSType),
 		},
 	}
+	// Send cpu arch attribute directly when running on-prem. When running on EC2, this is not needed
+	// since the cpu arch is reported via instance identity doc in that case.
+	if client.config.OnPrem.Enabled() {
+		attrs = append(attrs, &ecs.Attribute{
+			Name:  aws.String(cpuArchAttrName),
+			Value: aws.String(getCPUArch()),
+		})
+	}
+	return attrs
+}
+
+func getCPUArch() string {
+	arch := runtime.GOARCH
```

Comment:
does instance identity document expect just `arm` for all arm systems? do they not distinguish between arm 32- and 64-bit?

Reply:
according to this doc (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html) the IID returns `arm64`, whereas GOARCH would return just `arm`, I think we may also need to map arm -> arm64

Reply:
i think our tests only have amd64 and arm64. added unit test for the two. for 386, i don't think we have it in any of our existing platform but i added logic to change it to i386 just to be safe.